doc/languages-frameworks/agda.section.md

```diff
@@ -2,34 +2,126 @@
 
 ## How to use Agda
 
-Agda can be installed from `agda`:
+Agda is available as the [agda](https://search.nixos.org/packages?channel=unstable&show=agda&from=0&size=30&sort=relevance&query=agda)
+package and can be installed using `nix-env` although creating a `shell.nix` file in your project is preffered
+
 ```
-$ nix-env -iA agda
+{ nixpkgs ? <nixpkgs> }:
+with (import nixpkgs {});
+stdenv.mkDerivation {
```

Comment:
> I am unconvinced by having a tutorial for how to use nix-shell inside the agda documentation. I just feel that it makes it harder to actually access the information about agda?

I get your point. But on the other hand it turns out that installing Agda is not easy for many people. Nix could help with that. Therefore it would we nice to have a tutorial of some sort addressing readers new to Agda *and* Nix. Would something new like https://agda4nix.readthedocs.io/ make more sense to you?

Reply:
That, or an entry in agda.readthedocs.io. The nixpkgs manual should probably not start to introduce general concepts in the middle of a language specific section. But you can of course link to other parts in the manual that explains `nix-shell` and `mkShell` (or even write that part, if it's missing), and then go on using them here without introducing them here.

Reply:
I get the impression that the intended audience of people reading the language parts of nixpkgs manual are people who already use nix and what to know how to use language X in nix

Reply:
> That, or an entry in agda.readthedocs.io. The nixpkgs manual should probably not start to introduce general concepts in the middle of a language specific section. But you can of course link to other parts in the manual that explains `nix-shell` and `mkShell` (or even write that part, if it's missing), and then go on using them here without introducing them here.

Maybe referencing/improving the nixpkgs manual is the overall better approach. I'll dig into that.

Reply:
Removed the tutorialish part.